### PR TITLE
Remove exception checking from path query

### PR DESCRIPTION
**Pull request by duckofyork**

### CODE REVIEW

**Comment by duckofyork** [comment]
@aelred

**Comment by duckofyork** [comment]
!rtg

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/spanningtree/ChuLiuEdmonds.java [hunk 1]
```diff
@@ -46,7 +46,6 @@
  * Implementation is based on Tarjan's "Finding Optimum Branchings" paper:
  * http://cw.felk.cvut.cz/lib/exe/fetch.php/courses/a4m33pal/cviceni/tarjan-finding-optimum-branchings.pdf
  *
- * @author sthomson@cs.cmu.edu
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/spanningtree/ChuLiuEdmonds.java]: Why so mean? :(

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/PathQueryImpl.java [hunk 2]
```diff
@@ -76,15 +76,9 @@
...
-                    e.getCause().getMessage().equals(ErrorMessage.NO_PATH_EXIST.getMessage())) ||
-                    (e instanceof IllegalStateException &&
-                            e.getMessage().equals(ErrorMessage.NO_PATH_EXIST.getMessage()))) {
-                LOGGER.info("ShortestPathVertexProgram is done in " + (System.currentTimeMillis() - startTime) + " ms");
-                return Optional.empty();
-            }
-            throw e;
+        } catch (NoResultException e) {
+            LOGGER.info("ShortestPathVertexProgram is done in " + (System.currentTimeMillis() - startTime) + " ms");
+            return Optional.empty();
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/PathQueryImpl.java]: 👍

**Comment by duckofyork** [comment]
!rtg

**Comment by duckofyork** [comment]
@aelred This does not work on Janus...

**Comment by aelred** [comment]
@duckofyork 💔

**Pull request merged**